app/src/org/commcare/activities/FormAndDataSyncer.java

```diff
@@ -36,39 +36,29 @@
  */
 public class FormAndDataSyncer {
 
-    public FormAndDataSyncer() {
+    protected FormAndDataSyncer() {
     }
 
-    /**
-     * @return Were forms sent to the server by this method invocation?
-     */
-    public boolean checkAndStartUnsentFormsTask(SyncCapableCommCareActivity activity,
-                                                final boolean syncAfterwards,
-                                                boolean userTriggered) {
-        SqlStorage<FormRecord> storage = CommCareApplication.instance().getUserStorage(FormRecord.class);
-        FormRecord[] records = StorageUtils.getUnsentRecordsForCurrentApp(storage);
+
+    void startUnsentFormsTask(SyncCapableCommCareActivity activity,
+                              final boolean syncAfterwards,
+                              boolean userTriggered) {
 
         // We only want to update the last upload sync time when it's a blocking sync
         if (syncAfterwards) {
             HiddenPreferences.updateLastUploadSyncAttemptTime();
         }
 
-        if (records.length > 0) {
```

Comment:
made this code part of the [background task ](https://github.com/dimagi/commcare-android/blob/formSubmissionsScheduling/app/src/org/commcare/sync/FormSubmissionHelper.java#L79)